### ISSUE TITLE
Clippy `borrow_deref_ref` for `&PyAny` arguments

### ISSUE THREAD

**Issue by stuhood** [PyO3/pyo3#2555]
### Bug Description

For code like this:
https://github.com/pantsbuild/pants/blob/c69a70c3837e5fdecdae5d3a0475a5e6fdfcf20f/src/rust/engine/src/externs/mod.rs#L298

... `clippy` under Rust `1.63.0` renders an error like the following:
```
error: deref on an immutable reference
   --> src/externs/mod.rs:298:17
    |
298 |     input_arg0: &PyAny,
    |                 ^ help: if you would like to reborrow, try removing `&*`: `&`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#borrow_deref_ref
```

### Steps to Reproduce

1. Check out https://github.com/pantsbuild/pants/compare/main...stuhood:pants:stuhood/bump-to-rust-1.63.0-clippy-error
2. Run `./cargo clippy` (note the `./`, which refers to a wrapper script to set up env vars)

### Your operating system and version

macOS 11.6.8

### Your Python version (`python --version`)

Python 3.7

### Your Rust version (`rustc --version`)

1.63.0

### Your PyO3 version

0.16.5

### How did you install python? Did you use a virtualenv?

pyenv

**Comment by davidhewitt** [comment]
Ungh, yes, sorry about this. It's actually already fixed on `main` and I'm seeking to release 0.17 ASAP which will resolve (just working on a couple of other fixes to resolve at the same time). See #2493

**Comment by stuhood** [comment]
Got it: thanks a lot! The workaround (skipping this clippy lint) is not onerous, so don't rush on our account.
